packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -467,6 +525,21 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
   @override
   String formatYear(DateTime date) => date.year.toString();
 
+  @override
+  String formatCompactDate(DateTime date) {
```

Comment:
So what happens in other locales?  Is this just a fallback, or are the other locales just not implemented yet?

Reply:
This is just a fallback for the default case.

Other locales are handled by the _MaterialLocalizationsDelegate:

https://github.com/flutter/flutter/pull/50546/files/4e3dc0117f2531952d92b702b2e09db2aa9f82c6#diff-945add11ecf717a6dad5cb70168cd894R625

Which will use the `intl.DateFormat.yMd` for the given locale. This is how the other formats that were already there were implemented.